ext/sqlite3/sqlite3.c

```diff
@@ -1218,10 +1241,15 @@ PHP_METHOD(sqlite3, openBlob)
 
 	sqlite3_stream = emalloc(sizeof(php_stream_sqlite3_data));
 	sqlite3_stream->blob = blob;
+	sqlite3_stream->flags = flags;
 	sqlite3_stream->position = 0;
 	sqlite3_stream->size = sqlite3_blob_bytes(blob);
 
-	stream = php_stream_alloc(&php_stream_sqlite3_ops, sqlite3_stream, 0, "rb");
+	if (flags != 0) {
+		mode = "r+b";
+	}
```

Comment:
Hmm, having an int $flag parameter which works as boolean doesn't make much sense, IMHO. I'd either prefer a boolean $readwrite or maybe reusing `SQLITE3_OPEN_READONLY` and `SQLITE3_OPEN_READWRITE` here.